autoload/vimlparser.vim

```diff
@@ -3914,51 +3919,51 @@ function! s:Compiler.compile(node)
   if a:node.type == s:NODE_TOPLEVEL
     return self.compile_toplevel(a:node)
   elseif a:node.type == s:NODE_COMMENT
-    return self.compile_comment(a:node)
+    call self.compile_comment(a:node)
```

Comment:
```vim
if cond1
  "
elseif cond2
elseif cond3
elseif condN
endif
```
This is an issue of internal behavior of Vim. When not return early, Vim script must skip `if` conditions until `endfunction`.